### ISSUE TITLE
Impossible to save the MoViNet model.

### ISSUE THREAD

**Issue by AliKHaliliT** [tensorflow/models#10776]
Hello, 

First of all, I am talking about this [model](https://github.com/tensorflow/models/tree/master/official/projects/movinet).

I am sorry for my tone but, (and I am saving you the trouble of running the code and recreating the problem), it is impossible to save the model(Don't believe me, try it yourself and on GPU). I tried to fine-tune a A0 Base model with 2 classes and in fact I was successful (After lots and lots of improvisations of-course, why improvisations? try training it and you'll find out), but no matter what I tried, I was unable to save the model. Of-course, I can save the model but it skips some of the layers with a few errors like:

WARNING:tensorflow:Skipping full serialization of Keras layer <official.vision.modeling.layers.nn_layers.SpatialAveragePool3D object at 0x7f13ae981790>, because it is not built

The model itself is great, don't get me wrong. I achieved a very good accuracy but no matter how great the model is, it is useless if you can't practically use it. 

Anyways,
So when I try to load the saved model, I get various different errors. And I have tried all the possible ways to save as well as load the model but no luck. Also all the possible formats if you are wondering. 

The only way possible to save the model is to load it from the hub which basically is useless because you can't edit num_classes endpoint. And that is just with initializing the model and then saving it. I honestly don't know what will happen if I tried to train and then save it. 

The only possible working and useless way is to save the weights and then recreate the model later and load them back. Why useless? Well because you know... :)

Bottom line is, this model is currently useless and in my opinion, should be removed if the devs plan on doing nothing. 

Finally, I am not trying to disrespect anyone. I am just trying to state the facts and possibly get an answer to my problem, so if anybody knows how can I train this model and save it, please enlighten me :).
